lock/src/main/java/com/auth0/android/lock/PasswordlessLock.java

```diff
@@ -0,0 +1,285 @@
+            options.setAuthenticationParameters(defaultParams);
+        }
+
+        /**
+         * Finishes the construction of the Lock.Options and generates a new Lock instance
+         * with those Lock.Options.
+         *
+         * @return a new Lock instance configured as in the Builder.
+         */
+        public PasswordlessLock build() {
+            if (options.getAccount() == null) {
+                Log.e(TAG, "You need to specify the com.auth0.Auth0 object with the Auth0 Account details.");
+                throw new IllegalStateException("Missing Auth0 account information.");
+            }
+            if (callback == null) {
+                Log.e(TAG, "You need to specify the AuthenticationCallback object to receive the Authentication result.");
+                throw new IllegalStateException("Missing AuthenticationCallback.");
+            }
+            if (options.passwordlessMode() == null) {
+                Log.e(TAG, "You need to specify the PasswordlessMode to use in the Passwordless Authentication.");
+                throw new IllegalStateException("Missing PasswordlessMode.");
+            }
+            return new PasswordlessLock(options, callback);
+        }
+
+        /**
+         * Defines the Passwordless mode to use in the Authentication.
+         *
+         * @param mode a valid PasswordlessMode
+         * @return the current Builder instance
+         */
+        public Builder withMode(@NonNull PasswordlessMode mode) {
+            options.setPasswordlessMode(mode);
+            return this;
+        }
+
+        /**
+         * Whether the PasswordlessLockActivity can be closed when pressing the Back key or not.
+         *
+         * @param closable or not. By default, the LockActivity is not closable.
+         * @return the current builder instance
+         */
+        public Builder closable(boolean closable) {
+            options.setClosable(closable);
+            return this;
+        }
+
+        /**
+         * Whether the PasswordlessLockActivity will go fullscreen or will show the status bar.
+         *
+         * @param fullscreen or not. By default, the LockActivity will not be Fullscreen.
+         * @return the current builder instance
+         */
+        public Builder fullscreen(boolean fullscreen) {
+            options.setFullscreen(fullscreen);
+            return this;
+        }
+
+        /**
+         * Additional Authentication parameters can be set to use with different Identity Providers.
+         *
+         * @param authenticationParameters a non-null Map containing the parameters as Key-Values
+         * @return the current builder instance
+         */
+        public Builder withAuthenticationParameters(@NonNull Map<String, Object> authenticationParameters) {
+            if (authenticationParameters instanceof HashMap) {
+                options.setAuthenticationParameters((HashMap<String, Object>) authenticationParameters);
```

Comment:
Yes but this is conceptually wrong since we are accepting the generic interface and then force casting to a specific subclass, even if we have the `instanceof`. The better approach is to create a `HashMap` with the supplied map